blockchain/chain/src/store/chain_store.rs

```diff
@@ -994,28 +1022,29 @@ where
 #[cfg(feature = "json")]
 pub mod headchange_json {
     use super::*;
-    use blocks::tipset_json::TipsetJsonRef;
-    use serde::Serialize;
+    use blocks::tipset_json::TipsetJson;
+    use serde::{Deserialize, Serialize};
 
-    #[derive(Serialize)]
+    #[derive(Debug, Deserialize, Serialize)]
     #[serde(rename_all = "lowercase")]
     #[serde(tag = "type", content = "val")]
-    pub enum HeadChangeJson<'a> {
-        Current(TipsetJsonRef<'a>),
-        Apply(TipsetJsonRef<'a>),
-        Revert(TipsetJsonRef<'a>),
+    pub enum HeadChangeJson {
```

Comment:
Are you sure we cant do this by reference anymore?

Reply:
Is it better if we do it by reference? I took out the explicit lifetime because it didn't seem necessary anymore.

Reply:
Mmm, i think doing things by ref is better for serde serialization? I'm not sure though. Not a big concern for me. Just pointing it out.